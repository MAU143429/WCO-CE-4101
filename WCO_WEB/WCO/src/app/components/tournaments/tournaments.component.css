body{
  height: 100vh;
  justify-content: center;
  align-items: center;
  padding: 10px;
  background-color: white;

}
.container{
  max-width: 1700px;
  width: 100%;
  background-color: rgba(255, 255, 255, 0.295);
  padding: 25px 30px;
  border-radius: 5px;
  box-shadow: 0 5px 10px rgba(0,0,0,0.20);
}


.input-box{
  display: flex;
  margin-bottom: 15px;
  margin-left: 17%;
  width: 70%;
  font-size: 20px;
}

.title{
  width: 40%;
  font-size: 30px;
  font-weight: 500;
  margin-top: 1%;
  position: relative;
  justify-content: space-between;

}

.input-box input{
  height: 45px;
  width: 100%;
  font-size: 16px;
  border-radius: 5px;
  padding-left: 15px;
  border: 1px solid #ccc;
  border-bottom-width: 2px;
  transition: all 0.3s ease;
}


