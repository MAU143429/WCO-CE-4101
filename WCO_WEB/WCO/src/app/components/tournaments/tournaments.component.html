<!DOCTYPE html>
<html lang="en" dir="ltr">
  <head>
    <meta charset="UTF-8">
    <!---Referencia al archivo css que contiene el estilo--->
    <link rel="stylesheet" href="tournaments.component.css">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
   </head>
<!---Componentes que conforman la seccion de ver todos los torneos--->
<app-navbar></app-navbar>
<div class="input-box">
  <div class="title">Todos los torneos</div>
  <input  type="submit" value="Crear Torneo" routerLink="/create-tournaments">
</div>
<body>
  <div class="container">
    <div class="content">

      <table id="tournaments" class="table table-bordered mt-4 mytable">
        <tr class="table-primary mytable">
          <th>Nombre del Torneo</th>
          <th>ID</th>
          <th>Fecha de inicio</th>
          <th>Fecha de final</th>
          <th>Detalles</th>
        </tr>
      <tr *ngFor="let data of tournamentsData">
        <td>{{data.name}}</td>
          <td>{{data.toId}}</td>
          <td>{{data.startDate}}</td>
          <td>{{data.endDate}}</td>
          <td><button class="viewbtn" routerLink="/tournament-details" (click)="openTournament(data.toId)" >Ver Torneo</button></td>
      </tr>
      </table>
    </div>
  </div>
</body>
</html>


