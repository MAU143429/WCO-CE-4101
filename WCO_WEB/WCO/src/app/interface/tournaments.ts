import { DbBracket } from '../model/db-bracket';
import { DbTeam } from '../model/db-team';

export interface Tournaments {
  toId: string;
  name: string;
  startDate: string;
  endDate: string;
  type: string;
  description: string;
  teams: DbTeam[];
  brackets: DbBracket[];
}
