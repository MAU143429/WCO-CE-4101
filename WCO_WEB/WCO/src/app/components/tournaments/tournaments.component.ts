import { Component, Input, OnInit } from '@angular/core';
import { TournamentService } from '../../services/tournament.service';
import { InternalService } from '../../services/internal.service';
import { Tournaments } from '../../interface/tournaments';

@Component({
  selector: 'app-tournaments',
  templateUrl: './tournaments.component.html',
  styleUrls: ['./tournaments.component.css'],
})
export class TournamentsComponent implements OnInit {
  tournamentsData: Tournaments[] = [];
  constructor(
    private service: TournamentService,
    private connection: InternalService
  ) {}

  ngOnInit(): void {
    this.service.getTournaments().subscribe((data: Tournaments[]) => {
      this.tournamentsData = data;
    });
  }

  openTournament(id: string) {
    this.connection.setTournamentId(id);
    //this.connection.sendTournamentKey(id);
  }
}
