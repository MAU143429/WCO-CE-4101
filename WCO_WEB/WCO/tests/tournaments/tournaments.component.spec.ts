import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
import { ComponentFixture, TestBed } from '@angular/core/testing';
//import { http } from '@mobiscroll/angular/dist/js/core/util/http';
import { Observable, of } from 'rxjs';
import { TournamentService } from '../../src/app/services/tournament.service';

import { TournamentsComponent } from '../../src/app/components/tournaments/tournaments.component';

const dummyTournament = [{
  toId: '1',
  name: 'Mundial',
  startDate: '10/10/2022',
  endDate: '11/11/2022',
  type: 'Selecciones',
  description: 'Muy chiva',
  teams: [{teId: 0, name: 'Team1'}, {teId: 1, name: 'Team2'}],
  brackets: [{  
    bId: 0,
    name: 'Final',
    tounamentId: 'A2X'}]
}]

const mockTournamentService: {
  getTournaments: () => Observable<any>
} = {
  getTournaments: () => of(dummyTournament)
};


describe('TournamentsComponent', () => {
  let component: TournamentsComponent;
  let fixture: ComponentFixture<TournamentsComponent>;
  let compiled: HTMLElement;
  let httpMock: HttpTestingController;
  let service: TournamentService;


  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ TournamentsComponent ],
      imports: [
        HttpClientTestingModule
      ],
      providers: [
        {provide: TournamentService, useValue: mockTournamentService}
      ]
    })
    .compileComponents();
  
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(TournamentsComponent);
    component = fixture.componentInstance;
    service = TestBed.inject( TournamentService );
    httpMock = TestBed.inject( HttpTestingController );
    compiled = fixture.nativeElement;
    fixture.detectChanges();
  });

  

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe de hace match con el snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });

  test('En init debe traer torneo', () => {
    const setTournamentSpy = jest.spyOn(mockTournamentService, 'getTournaments')
    setTournamentSpy.mockReturnValue(of(dummyTournament));

    component.ngOnInit();
    expect(component.tournamentsData == dummyTournament)
  })

  test('openTorunament()', () => {
    component.openTournament("2")
    fixture.detectChanges();
    expect(localStorage.getItem('toID')).toEqual("2")
  });
});
