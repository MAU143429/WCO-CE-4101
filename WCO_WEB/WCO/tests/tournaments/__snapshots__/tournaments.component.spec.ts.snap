// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TournamentsComponent debe de hace match con el snapshot 1`] = `
"<html lang=\\"en\\" dir=\\"ltr\\"><head><meta charset=\\"UTF-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"></head><body><app-navbar></app-navbar><br><br><br><br><div class=\\"input-box\\"><div class=\\"title\\">Todos los torneos</div><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></div><div class=\\"container\\"><div class=\\"content\\"><table id=\\"tournaments\\" class=\\"table table-bordered mt-4 mytable\\"><tr class=\\"table-primary mytable\\"><th>Nombre del Torneo</th><th>ID</th><th>Fecha de inicio</th><th>Fecha de final</th><th>Detalles</th></tr><tr><td>Mundial</td><td>1</td><td>10/10/2022</td><td>11/11/2022</td><td><button routerlink=\\"/tournament-details\\" class=\\"viewbtn\\">Ver Torneo</button></td></tr><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"[object Object]\\"
}--></table></div></div></body></html>"
`;
